src/lib/hooks/hasConnectionChange.js

```diff
@@ -20,6 +21,57 @@ export const useConnection = () => {
   return isConnection
 }
 
+export const useConnectionWeb3 = () => {
+  const [isConnection, setIsConnection] = useState(true)
+  const store = SimpleStore.useStore()
+  const wallet = store.get('wallet')
+  const isWeb3Connection = () => {
+    if (wallet.wallet.currentProvider.connected) {
+      setIsConnection(true)
+    } else {
+      setIsConnection(false)
+      wallet.wallet.currentProvider.reconnect()
+      setTimeout(isWeb3Connection, 500)
+    }
+  }
+
+  useEffect(() => {
+    AppState.addEventListener('change', () => {
```

Comment:
does this happen on first app load? or only on focus?

Reply:
happen on first load too

Reply:
so obviously we need it to happen on first load